skbio/stats/distance/_base.py

```diff
@@ -366,6 +366,187 @@ def filter(self, ids, strict=True):
         filtered_data = self._data[idxs][:, idxs]
         return self.__class__(filtered_data, ids)
 
+    def _stable_order(self, ids):
+        """Obtain a stable ID order with respect to self
+
+        Parameters
+        ----------
+        ids : Iterable of ids
+            The IDs to establish a stable ordering for.
+
+        Returns
+        -------
+        np.array, dtype=int
+            The corresponding index values
+        """
+        # establish a stable ordering, and cache both the positions and label
+        # of each ID in the stable order.
+        id_order = sorted((self._id_index[i] for i in ids))
+        return np.array(id_order, dtype=int)
+
+    @experimental(as_of="0.5.4")
+    def within(self, ids):
+        """Obtain all the distances among the set of IDs
+
+        Parameters
+        ----------
+        ids : Iterable of str
+            The IDs to obtain distances for. All pairs of distances are
+            returned such that, if provided ['a', 'b', 'c'], the distances
+            for [('a', 'a'), ('a', 'b'), ('a', 'c'), ('b', 'a'), ('b', 'b'),
+            ('b', 'c'), ('c', 'a'), ('c', 'b'), ('c', 'c')] are gathered.
+
+        Returns
+        -------
+        pd.DataFrame
+            (i, j, value) representing the source ID ("i"), the target ID ("j")
+            and the distance ("value").
+
+        Raises
+        ------
+        MissingIDError
+            If an ID(s) specified is not in the dissimilarity matrix.
+
+        Notes
+        -----
+        Order of the return items is stable, meaning that requesting IDs
+        ['a', 'b'] is equivalent to ['b', 'a']. The order is with respect
+        to the order of the .ids attribute of self.
+
+        Example
+        -------
+        >>> from skbio.stats.distance import DissimilarityMatrix
+        >>> dm = DissimilarityMatrix([[0, 1, 2, 3, 4], [1, 0, 1, 2, 3],
+        ...                           [2, 1, 0, 1, 2], [3, 2, 1, 0, 1],
+        ...                           [4, 3, 2, 1, 0]],
+        ...                          ['A', 'B', 'C', 'D', 'E'])
+        >>> dm.within(['A', 'B', 'C'])
+           i  j  value
+        0  A  A    0.0
+        1  A  B    1.0
+        2  A  C    2.0
+        3  B  A    1.0
+        4  B  B    0.0
+        5  B  C    1.0
+        6  C  A    2.0
+        7  C  B    1.0
+        8  C  C    0.0
+        """
+        ids = set(ids)
+
+        if ids - set(self._id_index):
+            missing = list(ids - set(self._id_index))
+            raise MissingIDError("At least one ID (e.g., '%s') was not "
+                                 "found." % missing[0])
+
+        i_order = self._stable_order(ids)
+        return self._subset_to_dataframe(i_order, i_order)
+
+    @experimental(as_of="0.5.4")
+    def between(self, from_, to_, allow_overlap=False):
+        """Obtain all the distances among the set of IDs
```

Comment:
Doesn't this function 'Obtain all the distances from `from_` to `to_`' or less precisely, 'Obtain all the distances between the sets of IDS'?